server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -254,6 +257,7 @@ public DesiredBalance compute(
         final long computationStartedTime = threadPool.relativeTimeInMillis();
         long nextReportTime = computationStartedTime + timeWarningInterval;
 
+        clusterInfoSimulator.discardReservedSpace();
```

Comment:
The `DesiredBalanceComputerTests` all pass if this line is removed, which seems suspicious. Could we have a test that exercises this?